src/main/java/WayofTime/bloodmagic/item/sigil/ItemSigilLava.java

```diff
@@ -34,80 +41,61 @@ public ItemSigilLava() {
 
         if (!world.isRemote && !isUnusable(stack)) {
             RayTraceResult rayTrace = this.rayTrace(world, player, false);
-
-            if (rayTrace != null) {
-                ActionResult<ItemStack> ret = ForgeEventFactory.onBucketUse(player, world, stack, rayTrace);
-                if (ret != null)
-                    return ret;
-
-                if (rayTrace.typeOfHit == RayTraceResult.Type.BLOCK) {
-                    BlockPos blockpos = rayTrace.getBlockPos();
-
-                    if (!world.isBlockModifiable(player, blockpos)) {
-                        return super.onItemRightClick(world, player, hand);
-                    }
-
-                    if (!player.canPlayerEdit(blockpos.offset(rayTrace.sideHit), rayTrace.sideHit, stack)) {
-                        return super.onItemRightClick(world, player, hand);
-                    }
-
-                    BlockPos blockpos1 = blockpos.offset(rayTrace.sideHit);
-
-                    if (!player.canPlayerEdit(blockpos1, rayTrace.sideHit, stack)) {
-                        return super.onItemRightClick(world, player, hand);
-                    }
-
-                    if (canPlaceLava(world, blockpos1) && NetworkHelper.getSoulNetwork(getBinding(stack)).syphonAndDamage(player, SoulTicket.item(stack, world, player, getLpUsed())).isSuccess() && tryPlaceLava(world, blockpos1)) {
-                        return super.onItemRightClick(world, player, hand);
+            
+            ActionResult<ItemStack> ret = ForgeEventFactory.onBucketUse(player, world, stack, rayTrace);
+            if (ret != null) return ret;
```

Comment:
I've never actually had the onBucketUse event thing fire in all of my testing, since it still runs off to the actual FluidUtil#tryPlaceFluid later on and places the fluid with the syphon code. I'll remove that snippet just to be sure though.

If I can't implement IFluidHandlerItem due to the fact that actually giving the sigils the correct hasCapability causes their ability to syphon on draining to actually break (due to the game apparently favoring running the tile entity's on click handlers it seems, at least when you're not shift clicking), then I'll have to dump everything that uses the FluidUtil methods and essentially re-implement the ones each sigil uses in the sigils themselves (with a few modifications like not trying to drain from the sigils).
So if the standard is strict and I definitely should not implement IFluidHandlerItem without also declaring hasCapability/getCapability for FLUID_HANDLER_ITEM_CAPABILITY, I will do so. Just wanna be sure that that's the only issue at hand and I'm not missing something else on that subject.